generator/transit_generator.cpp

```diff
@@ -43,20 +60,79 @@ string GetFileName(string const & filePath)
   return name;
 }
 
-/// \brief Reads from |root| (json) and serializes an array to |serializer|.
-/// \param handler is function which fixes up vector of |Item|(s) after deserialization from json
-/// but before serialization to mwm.
 template <class Item>
-void SerializeObject(my::Json const & root, string const & key, Serializer<FileWriter> & serializer,
-                     function<void(vector<Item> &)> handler = nullptr)
+void DeserializeFromJson(my::Json const & root, string const & key, vector<Item> & items)
 {
-  vector<Item> items;
-
+  items.clear();
   DeserializerFromJson deserializer(root.get());
   deserializer(items, key.c_str());
+}
+
+void DeserializeGatesFromJson(my::Json const & root, string const & mwmDir, string const & countryId,
+                              vector<Gate> & gates)
+{
+  DeserializeFromJson(root, "gates", gates);
+
+  // Creating IndexRouter.
+  LocalCountryFile localCountryFile;
+  Index index;
```

Comment:
My proposal was to create a small wrapper around indexer, say:
```c++
class SingleMwmIndex
{
public:
  SingleMwmIndex(string const & path) { ... }

private:
  Index m_index;
};
```
because you still need to duplicate declarations of index and result.

Reply:
Yes. It's better. Let's do something like that:
```c++
class SingleMwmIndex
{
public:
  SingleMwmIndex(string const & path) { ... }

private:
  Index m_index;
  LocalCountryFile m_countryFile;
};
```
because we need country file sometimes.